ocean_provider/validation/algo.py

```diff
@@ -145,15 +146,7 @@ def _build_and_validate_algo(self, algo_data):
                 return False
 
             try:
-                dt = get_datatoken_contract(self.web3, self.consumer_address)
-                tx_receipt = get_tx_receipt(self.web3, algorithm_tx_id)
-                event_logs = dt.events.OrderStarted().processReceipt(
-                    tx_receipt, errors=DISCARD
-                )
-
-                order_log = event_logs[0] if event_logs else None
-                algo_service_index = order_log.args.serviceIndex
-                self.algo_service = algo.get_service_by_index(algo_service_index)
+                self.algo_service = algo.get_service_by_id(algorithm_service_id)
```

Comment:
please keep the order check, we need to check that the service was ordered. Retrieve the algorithm service based on the provided argument, then check that the algorithm_tx_id indeed reflects that same algorithm Service. If it does not, add the error that the algo tx id and algo service id do not correspond. And please also add a test for this.

Reply:
The order check is performed here, if I'm not wrong: https://github.com/oceanprotocol/provider/blob/d1fafca4a1dbbe1af1d0a0a8abe3ec204be54ad9/ocean_provider/validation/algo.py#L164